### ISSUE TITLE
proxy-url is not supported in the KUBECONFIG

### ISSUE THREAD

**Issue by ktarplee** [davidB/kubectl-view-allocations#235]
I am using version 0.18.1.

```txt
kubectl view-allocations 
2024-04-15T17:04:24.958680Z ERROR failed with error error trying to connect: dns error: failed to lookup address information: Name does not resolve
Error: 
   0: Failed to list nodes
   1: HyperError: error trying to connect: dns error: failed to lookup address information: Name does not resolve
   2: error trying to connect: dns error: failed to lookup address information: Name does not resolve
   3: dns error: failed to lookup address information: Name does not resolve
   4: failed to lookup address information: Name does not resolve

Location:
   src/main.rs:47
```
The relevant part of my kubeconfig is
```yaml
clusters:
- cluster:
    certificate-authority-data: OMITTED
    server: https://ctl.ktarplee:6443
    proxy-url: socks5://localhost:9999 # don't forget to run "ssh -D 9999 head"
  name: ktarplee
```

`ctl.ktarplee` will resolve and can be connected to only over the `proxy-url`.  Just run `ssh -D 9999 head` where `head` is some node on your cluster.  Then you will see that `kubectl`, `helm`, etc. all work just fine.  However `kubectl view-allocations` does not work.  See the Kubernetes docs about this [here](https://kubernetes.io/docs/tasks/extend-kubernetes/socks5-proxy-access-api/).

This is likely an issue in a dependency of this project.

**Comment by ktarplee** [comment]
@davidB It looks like you are on kube=0.87.  `kube` version 0.87.1 (see the changes [here](https://github.com/kube-rs/kube/releases/tag/0.87.1)) supports SOCKS5 and proxy-url.  So it might be as simple as updating that dependency.

**Comment by davidB** [comment]
The fix is in the version 0.19.2
